### PR TITLE
[Mentions v2] Add support for <mention-report /> tag

### PR DESCRIPTION
**Pull request by robertKozik**
### Details
This PR is part of [Mentions v2 design doc](https://docs.google.com/document/d/1g_1xLGO29EV6hbOtiYqX53oof_gr8jbnbkSRPjuA4RI/edit). It adds support for new kind of mention: **room mention** which is described with `<mention-report />` HTML tag. This mention should be styled the same as user mention
image

### Related Issues

https://github.com/Expensify/App/issues/39542

### Manual Tests

Type room mention (starts with `#` followed by any combination of small letters, numbers and hyphens ) and verify that

### CODE REVIEW

File: android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java [hunk 1]
```diff
@@ -124,6 +124,11 @@ private void applyRange(SpannableStringBuilder ssb, String type, int start, int
         setSpan(ssb, new MarkdownForegroundColorSpan(mMarkdownStyle.getMentionUserColor()), start, end);
         setSpan(ssb, new MarkdownBackgroundColorSpan(mMarkdownStyle.getMentionUserBackgroundColor()), start, end);
         break;
+      case "mention-report":
+        // TODO: change mention color when it mentions current user
```
Comment (tomekzaw) [inline android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java:128]: Is this comment relevant in this case?

File: ios/RCTMarkdownStyle.mm [hunk 2]
```diff
@@ -41,6 +41,9 @@ - (instancetype)initWithStruct:(const facebook::react::MarkdownTextInputDecorato
 
     _mentionUserColor = RCTUIColorFromSharedColor(style.mentionUser.color);
     _mentionUserBackgroundColor = RCTUIColorFromSharedColor(style.mentionUser.backgroundColor);
+      
+  _mentionReportColor = RCTUIColorFromSharedColor(style.mentionReport.color);
+  _mentionReportBackgroundColor = RCTUIColorFromSharedColor(style.mentionReport.backgroundColor);
```
Comment (tomekzaw) [inline ios/RCTMarkdownStyle.mm:46]: Let's adjust the indent here and remove unnecessary spaces in the blank line
```suggestion

    _mentionReportColor = RCTUIColorFromSharedColor(style.mentionReport.color);
    _mentionReportBackgroundColor = RCTUIColorFromSharedColor(style.mentionReport.backgroundColor);
```

File: ios/RCTMarkdownUtils.mm [hunk 3]
```diff
@@ -108,6 +108,10 @@ - (NSAttributedString *)parseMarkdown:(nullable NSAttributedString *)input withA
                 // TODO: change mention color when it mentions current user
                 [attributedString addAttribute:NSForegroundColorAttributeName value:_markdownStyle.mentionUserColor range:range];
                 [attributedString addAttribute:NSBackgroundColorAttributeName value:_markdownStyle.mentionUserBackgroundColor range:range];
+            } else if ([type isEqualToString:@"mention-report"]) {
+                // TODO: change mention color when it mentions current user
```
Comment (tomekzaw) [inline ios/RCTMarkdownUtils.mm:112]: same as above

**Comment by tomekzaw** [comment]
please upgrade screenshot in PR description as well

**Review by tomekzaw — APPROVED** [review]
Thanks for this PR! 👍

**Review by rlinoz — APPROVED** [review]
LGTM!

**Comment by rlinoz** [comment]
Are this jobs stuck or is that normal?

**Comment by robertKozik** [comment]
It takes lot of time 😅 I'll proceed with merging

**Pull request merged**